controllers/provisioners/eksfargate/eksfargate.go

```diff
@@ -0,0 +1,262 @@
+)
+
+var log = logrus.New()
+
+const LastAppliedConfigurationKey = "kubectl.kubernetes.io/last-applied-configuration"
+
+const (
+	PendingRolePolicyAttach = "pendingPolicyCreation"
+	PendingRoleCreation     = "pendingRoleCreation"
+)
+
+const (
+	OngoingStateString             = "OngoingState"
+	FiniteStateString              = "FiniteState"
+	FiniteDeletedString            = "FiniteDeleted"
+	UpdateRecoverableErrorString   = "UpdateRecoverableError"
+	UnrecoverableErrorString       = "UnrecoverableError"
+	UnrecoverableDeleteErrorString = "UnrecoverableDeleteError"
+)
+
+func New(instanceGroup *v1alpha1.InstanceGroup, worker *aws.AwsFargateWorker) (*InstanceGroupContext, error) {
+	ctx := InstanceGroupContext{
+		InstanceGroup:    instanceGroup,
+		AwsFargateWorker: worker,
+	}
+	instanceGroup.SetState(v1alpha1.ReconcileInit)
+	return &ctx, nil
+}
+func CreateFargateTags(tagArray []map[string]string) map[string]*string {
+	tags := make(map[string]*string)
+	for _, t := range tagArray {
+		for k, v := range t {
+			va := new(string)
+			*va = v
+			tags[k] = va
+		}
+	}
+	return tags
+}
+
+// Convienence function to convert from json to API.
+func CreateFargateSelectors(selectors []*v1alpha1.EKSFargateSelectors) []*eks.FargateProfileSelector {
+	var eksSelectors []*eks.FargateProfileSelector
+	for _, selector := range selectors {
+		m := make(map[string]*string)
+		for k, v := range selector.Labels {
+			vv := new(string)
+			*vv = v
+			m[k] = vv
+		}
+		eksSelectors = append(eksSelectors, &eks.FargateProfileSelector{Namespace: selector.Namespace, Labels: m})
+	}
+	return eksSelectors
+}
+
+func (ctx *InstanceGroupContext) Create() error {
+	log.SetFormatter(&logrus.TextFormatter{
+		FullTimestamp: true,
+	})
+	var arn *string
+	instanceGroup := ctx.GetInstanceGroup()
+	if *instanceGroup.Spec.EKSFargateSpec.GetPodExecutionRoleArn() == "" {
+		created, err := ctx.AwsFargateWorker.CreateDefaultRole()
+		if created || err != nil {
+			log.Infof("Creating default role: %v", err)
+			return err
+		}
+		role, err := ctx.AwsFargateWorker.GetDefaultRole()
+		if err != nil {
+			log.Errorf("Failed to get default role: %v", err)
+			return err
+		}
+		arn = role.Arn
+
+		err = ctx.AwsFargateWorker.AttachDefaultPolicyToDefaultRole()
+		if err != nil {
+			log.Errorf("Failed to get attach policy to role: %v", err)
+			return err
+		}
+		log.Info("Attached default policy to role")
+
+	} else {
+		arn = instanceGroup.Spec.EKSFargateSpec.GetPodExecutionRoleArn()
+	}
+
+	log.Infof("Creating a profile with %s", *arn)
+	err := ctx.AwsFargateWorker.CreateProfile(arn)
+	if err != nil {
+		log.Errorf("Creation of the fargate profile for cluster %v and name %v failed: %v",
+			*instanceGroup.Spec.EKSFargateSpec.GetClusterName(),
+			*instanceGroup.Spec.EKSFargateSpec.GetProfileName(),
+			err)
+	} else {
+		instanceGroup.SetState(v1alpha1.ReconcileModifying)
+	}
+
+	return err
+}
+func (ctx *InstanceGroupContext) CloudDiscovery() error {
+	return nil
```

Comment:
You should move your AWS logic/state here.

Reply:
will do

Reply:
done